File: 231. Power of Two/main.cpp
// Example 1:
// Input: n = 1
// Output: true
// Explanation: 2^0 = 1

// Example 2:
// Input: n = 16
// Output: true
// Explanation: 2^4 = 16

// Example 3:
// Input: n = 3
// Output: false

#include <iostream>
#include <cmath>

using namespace std;

int main()
{
    int num;
    cout << "Enter a number";
    cin >> num;
    bool found = false;

    for (int i = 0; i <= 30; i++)
    {
        int power = pow(2, i);
        if (num == power)
        {
            found = true;
        }
    }

    if (found)
    {
        cout << "Found";
    }
    else
    {
        cout << "Not Found";
    }
}

File: 231. Power of Two/main.rs
use std::io;

fn main() {
    let mut inp = String::new();
    println!("Please Enter a number: ");

    io::stdin()
        .read_line(&mut inp)
        .expect("Failed to take input");

    let num: i32 = match inp.trim().parse() {
        Ok(n) => n,
        Err(_) => {
            println!("Invalid input, please enter a valid integer.");
            return;
        }
    };

    // let ans: i32 = 1;
    let mut flag: bool = false;

    for i in 0..31 {
        if 2_i32.pow(i) == num {
            flag = true;
            break;
        }
    }

    match flag {
        true => println!("Found"),
        false => println!("Not Found"),
    }
}
